java-checks/src/test/java/org/sonar/java/checks/helpers/ValueBasedUtilsTest.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.checks.helpers;
+
+import com.google.common.base.Charsets;
+import com.google.common.collect.ImmutableList;
+import java.io.File;
+import java.util.List;
+import org.fest.assertions.BooleanAssert;
+import org.junit.Test;
+import org.sonar.java.ast.parser.JavaParser;
+import org.sonar.java.resolve.SemanticModel;
+import org.sonar.plugins.java.api.tree.ClassTree;
+import org.sonar.plugins.java.api.tree.CompilationUnitTree;
+import org.sonar.plugins.java.api.tree.Tree;
+import org.sonar.plugins.java.api.tree.VariableTree;
+
+import static org.fest.assertions.Assertions.assertThat;
+
+public class ValueBasedUtilsTest {
+
+  @Test
+  public void testIsValueBased() throws Exception {
+    File file = new File("src/test/files/checks/helpers/ValueBasedUtilsTest.java");
+    CompilationUnitTree tree = (CompilationUnitTree) JavaParser.createParser(Charsets.UTF_8).parse(file);
+    SemanticModel.createFor(tree, ImmutableList.<File>of());
+    List<Tree> members = null;
+    
+    members = ((ClassTree) tree.types().get(0)).members();
+    assertValueBased(members, 0).isFalse();
+    assertValueBased(members, 1).isTrue();
+    assertValueBased(members, 2).isTrue();
+    assertValueBased(members, 3).isFalse();
+    assertValueBased(members, 4).isTrue();
+    assertValueBased(members, 5).isFalse();
+    assertValueBased(members, 6).isFalse();
```

Comment:
Index based test cases is not the best thing for maintainance. It could be easier to rely on comments within the file : those can be retrieved using firstToken.getTrivias and check the content of this comment. This allow refactoring of the test file without pain for maintainance.